tests/cpp/merge_mps/merge_mps_test.cpp

```diff
@@ -0,0 +1,213 @@
+        std::ofstream structure_file(options_.STRUCTURE_FILE);
+        for (const auto &[mps, var, idx]: entries) {
+            structure_file << fmt::format("{0} {1} {2}\n", mps, var, idx);
+        }
+        structure_file.close();
+    }
+
+    std::filesystem::path tmp_dir_;
+    MergeMPSOptions options_;
+    std::shared_ptr<Xpansion::Test::LoggerNOOPStub> logger_;
+    std::shared_ptr<Xpansion::Test::InMemoryWriter> writer_;
+};
+
+TEST(MergeMPS, empty_input_ok) {
+    auto tmpDir = CreateRandomSubDir(std::filesystem::temp_directory_path());
+    std::ofstream structure_file(tmpDir / "structure_file.txt"s);
+
+    MergeMPSOptions options;
+    options.SOLVER_NAME = "COIN";
+    options.STRUCTURE_FILE = (tmpDir / "structure_file.txt"s).string();
+    auto logger = std::make_shared<Xpansion::Test::LoggerNOOPStub>();
+    auto writer = std::make_shared<Xpansion::Test::InMemoryWriter>();
+    std::filesystem::current_path(tmpDir);
+    MergeMPS mergeMPS(options, logger, writer);
+    mergeMPS.launch();
+    auto lastSolution = writer->solution_data_;
+    EXPECT_EQ(lastSolution.problem_status, "ERROR");
+}
+
+TEST_F(MergeMPSTest, one_master_Problem_ok) {
+    createMasterProblem();
+    createStructureFile({{"master.mps", "X1", 0}, {"master.mps", "X2", 1}});
+    std::filesystem::current_path(tmp_dir_);
+    MergeMPS mergeMPS(options_, logger_, writer_);
+    mergeMPS.launch();
+    auto lastSolution = writer_->solution_data_;
+    EXPECT_EQ(lastSolution.problem_status, "OPTIMAL");
+    //log_merged_mps exists
+    std::cout << tmp_dir_ / "log_merged.mps" << std::endl;
+    EXPECT_TRUE(std::filesystem::exists(tmp_dir_ / "log_merged.mps"s));
+    EXPECT_TRUE(std::filesystem::exists(tmp_dir_ / "log_merged.lp"s));
+    //Print the merged problem
+    std::ifstream merged_mps(tmp_dir_ / "log_merged.mps"s);
+    std::string line;
+    while (std::getline(merged_mps, line)) {
+        std::cout << line << std::endl;
+    }
+    merged_mps.close();
+    SolverFactory factory;
+    auto merged = factory.create_solver("CBC");
+    merged->read_prob_lp(tmp_dir_ / "log_merged.lp"s);
+    auto master = factory.create_solver("CBC");
+    master->read_prob_mps(tmp_dir_ / "master.mps"s);
+    //Merged and master problems are identical
+    EXPECT_EQ(merged->get_ncols(), master->get_ncols());
+    EXPECT_EQ(merged->get_nrows(), master->get_nrows());
+    EXPECT_EQ(merged->get_nelems(), master->get_nelems());
+    //Merged and master problems have the same objective function
+    DblVector obj_merged(merged->get_ncols());
+    DblVector obj_master(master->get_ncols());
+    merged->get_obj(obj_merged.data(), 0, merged->get_ncols() - 1);
+    master->get_obj(obj_master.data(), 0, master->get_ncols() - 1);
+    for (int i = 0; i < merged->get_ncols(); ++i) {
+        EXPECT_DOUBLE_EQ(obj_merged[i], obj_master[i]);
+    }
+    //BOth have the same solution
+    DblVector sol_merged(merged->get_ncols());
+    DblVector sol_master(master->get_ncols());
```

Comment:
Same question for solution and any other properties from the solver we want to check